examples/api/test/material/selection_area/selection_area.0_test.dart

```diff
@@ -0,0 +1,74 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui';
+
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_api_samples/material/selection_area/selection_area.0.dart'
+    as example;
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  TestWidgetsFlutterBinding.ensureInitialized();
+  final _MockClipboard mockClipboard = _MockClipboard();
+
+  testWidgets('SelectionArea smoke test', (WidgetTester tester) async {
+    tester.binding.defaultBinaryMessenger.setMockMethodCallHandler(
+      SystemChannels.platform,
+      mockClipboard.handleMethodCall,
+    );
+
+    await tester.pumpWidget(
+      const example.SelectionAreaExampleApp(),
+    );
+
+    expect(find.byType(SelectionArea), findsExactly(1));
+
+    Finder finder = find.descendant(
+      of: find.byType(SelectionArea),
+      matching: find.descendant(
+        of: find.byType(AppBar),
+        matching: find.text('SelectionArea Sample'),
+      ),
+    );
+
+    expect(finder, findsExactly(1));
+    expect(mockClipboard.clipboardData, isNull);
+
+    finder = find.text('SelectionArea Sample');
+    final Rect rect = tester.getRect(finder);
+
+    await tester.dragFrom(
+      Offset(rect.left, rect.top),
+      Offset(rect.right, rect.bottom),
+      kind: PointerDeviceKind.mouse,
+    );
+    await tester.pumpAndSettle();
+    await tester.sendKeyDownEvent(LogicalKeyboardKey.control);
+    await tester.sendKeyDownEvent(LogicalKeyboardKey.keyC);
+    await tester.sendKeyUpEvent(LogicalKeyboardKey.control);
+    await tester.sendKeyUpEvent(LogicalKeyboardKey.keyC);
+
+    expect(mockClipboard.clipboardData?['text'], 'SelectionArea Sample');
```

Comment:
Checking both that the text is inside the SelectionArea and that it can be copied (by simulating the Ctrl+C) seems redundant to me.

I would consider only checking that texts (the title but also other texts from the sample code) are part of the SelectionArea. My motivations are:
- the fact that Ctrl+C is working is already tested in the `SelectionArea` main tests.
- it will simplify this test and narrow it to what the example is doing.

Reply:
Thanks for the detailed explanation. I will make the necessary changes.